enos/enos-scenario-replication.hcl

```diff
@@ -0,0 +1,667 @@
+    }
+  }
+
+  step "get_secondary_cluster_ips" {
+    module     = module.vault_cluster_ips
+    depends_on = [step.verify_vault_secondary_unsealed]
+
+    providers = {
+      enos = local.enos_provider[matrix.distro]
+    }
+
+    variables {
+      vault_instances   = step.create_vault_secondary_cluster.vault_instances
+      vault_install_dir = local.vault_install_dir
+      vault_root_token  = step.create_vault_secondary_cluster.vault_root_token
+    }
+  }
+
+  step "verify_vault_primary_write_data" {
+    module     = module.vault_verify_write_data
+    depends_on = [step.get_primary_cluster_ips]
+
+
+    providers = {
+      enos = local.enos_provider[matrix.distro]
+    }
+
+    variables {
+      leader_public_ip  = step.get_primary_cluster_ips.leader_public_ip
+      leader_private_ip = step.get_primary_cluster_ips.leader_private_ip
+      vault_instances   = step.create_vault_primary_cluster.vault_instances
+      vault_install_dir = local.vault_install_dir
+      vault_root_token  = step.create_vault_primary_cluster.vault_root_token
+    }
+  }
+
+  step "configure_performance_replication_primary" {
+    module = module.vault_performance_replication_primary
+    depends_on = [
+      step.get_primary_cluster_ips,
+      step.verify_vault_primary_write_data
+    ]
+
+    providers = {
+      enos = local.enos_provider[matrix.distro]
+    }
+
+    variables {
+      primary_leader_public_ip  = step.get_primary_cluster_ips.leader_public_ip
+      primary_leader_private_ip = step.get_primary_cluster_ips.leader_private_ip
+      vault_install_dir         = local.vault_install_dir
+      vault_root_token          = step.create_vault_primary_cluster.vault_root_token
+    }
+  }
+
+  step "generate_secondary_token" {
+    module     = module.generate_secondary_token
+    depends_on = [step.configure_performance_replication_primary]
+
+    providers = {
+      enos = local.enos_provider[matrix.distro]
+    }
+
+    variables {
+      primary_leader_public_ip = step.get_primary_cluster_ips.leader_public_ip
+      vault_install_dir        = local.vault_install_dir
+      vault_root_token         = step.create_vault_primary_cluster.vault_root_token
+    }
+  }
+
+  step "configure_performance_replication_secondary" {
+    module     = module.vault_performance_replication_secondary
+    depends_on = [step.generate_secondary_token]
+
+    providers = {
+      enos = local.enos_provider[matrix.distro]
+    }
+
+    variables {
+      secondary_leader_public_ip  = step.get_secondary_cluster_ips.leader_public_ip
+      secondary_leader_private_ip = step.get_secondary_cluster_ips.leader_private_ip
+      vault_install_dir           = local.vault_install_dir
+      vault_root_token            = step.create_vault_secondary_cluster.vault_root_token
+      wrapping_token              = step.generate_secondary_token.secondary_token
+    }
+  }
+
+  step "unseal_secondary_followers" {
```

Comment:
Can we add a comment on which steps are workarounds for bugs here as well as in the called modules?

Reply:
The step itself is not a workaround, the method used within it is.. so added the comment within the module